src/agent/rustjail/src/mount.rs

```diff
@@ -1050,10 +1055,37 @@ mod tests {
     use std::fs::create_dir;
     use std::fs::create_dir_all;
     use std::fs::remove_dir_all;
+    use std::io;
     use std::os::unix::fs;
     use std::os::unix::io::AsRawFd;
     use tempfile::tempdir;
 
+    // Parameters:
+    //
+    // 1: expected Result
+    // 2: actual Result
+    // 3: string used to identify the test on error
+    macro_rules! assert_result {
+        ($expected_result:expr, $actual_result:expr, $msg:expr) => {
+            if $expected_result.is_ok() {
+                let expected_level = $expected_result.as_ref().unwrap();
+                let actual_level = $actual_result.unwrap();
+                assert!(*expected_level == actual_level, "{}", $msg);
+            } else {
+                let expected_error = $expected_result.as_ref().unwrap_err();
+                let expected_error_msg = format!("{:?}", expected_error);
+
+                if let Err(actual_error) = $actual_result {
+                    let actual_error_msg = format!("{:?}", actual_error);
+
+                    assert!(expected_error_msg == actual_error_msg, "{}", $msg);
+                } else {
+                    assert!(expected_error_msg == "expected error, got OK", "{}", $msg);
```

Comment:
This line is something tricky.  It's better to use `assert!(result.is_err())` to check error result. For example, how about this:

```rust
assert!($actual_result.is_err());

let actual_error_msg = format!("{:?}", $actual_result.unwrap_err());
assert!(expected_error_msg == actual_error_msg, "{}", $msg);
```

Reply:
This `assert_result` macro is interesting, it's been useful so it is now copied in a few different places. See [this comment](https://github.com/kata-containers/kata-containers/pull/3838/files#r822590397) from James.

I'll incorporate your feedback in this PR though, and probably in all places the `assert_result` macro is defined so they stay in sync.

Reply:
@bradenrayhorn you can create an issue to track this, and put them in `agent/src/test_utils.rs` if possible.

Reply:
Thank you @liubin, I did not realize this file existed. I created #4093. Did you want this change in this PR? Or should I leave `assert_result` as-is in this PR, and open a future PR to address #4093?

Reply:
@bradenrayhorn I think both are fine.

Reply:
@liubin I've moved the `assert_result` macro to where the other `rustjail` common test macros are. Because `agent/src/test_utils.rs` is in a different crate, I don't think we can have rustjail importing in, but please correct me if I am wrong.

I will open a separate PR to add `assert_result` to the `agent/src/test_utils.rs` file for use in the agent crate tests.